main/inc/lib/myspace.lib.php

```diff
@@ -1156,137 +1156,253 @@ public static function displayResumeLP(
-            INNER JOIN $tblCourse ON $tblCourse.id = $lp_table.c_id
-            WHERE
-                $lp_table.iid = $lp_id
-                ";
-                $res = Database::query($query);
-                if (Database::num_rows($res)) {
-                    $lpInfo = Database::fetch_array($res);
+        $dataTeachers = Database::store_result($queryResult, 'ASSOC');
+        $totalData = count($dataTeachers);
+        $lpItems = [];
+        $teachers = [];
+        $users = [];
+        $learningPaths = [];
+        $csvContent = [];
+        $htmlData = '';
+
+        for ($i = 0; $i < $totalData; $i++) {
+            $row = $dataTeachers[$i];
+            $lpId = $row['lp_id'];
+            $lpItems[] = $lpId;
+            $authorData = $row['users_id'];
+            $learningPaths[$lpId] = $row;
+            if (strpos($authorData, ",") === false) {
+                if (!isset($users[$authorData])) {
+                    $users[$authorData] = api_get_user_info($authorData);
                 }
-                $studentUsers = [];
-                for ($i = 0; $i < count($registeredUsers); $i++) {
-                    $studentUsers[] = api_get_user_info($registeredUsers[$i]);
-                }
-                $teacherList = [];
-                for ($i = 0; $i < count($teachersId); $i++) {
-                    $teacherId = $teachersId[$i];
-                    $teacher = api_get_user_info($teacherId);
-                    $data[$teacher['complete_name']][$lpInfo['name']] = [
-                        'students' => count($studentUsers),
-                        'studentList' => $studentUsers,
-                        'lpInfo' => $lpInfo,
-                    ];
-                    $teacherList[] = $teacher;
+                $teachers[$authorData][$lpId] = $users[$authorData];
+                $learningPaths[$lpId]['teachers'][$authorData] = $users[$authorData];
+            } else {
+                $items = explode(',', $authorData);
+                for ($j = 0; $j < count($items); $j++) {
+                    $authorData = $items[$j];
+                    if (!isset($users[$authorData])) {
+                        $users[$authorData] = api_get_user_info($authorData);
+                    }
+                    $teachers[$authorData][$lpId] = $users[$authorData];
+                    $learningPaths[$lpId]['teachers'][$authorData] = $users[$authorData];
                 }
             }
         }
-        if ($csv == false) {
-            $table = "<div class='table-responsive'>".
+        $lpItems = array_unique($lpItems);
+        $whereInLp = implode(',', $lpItems);
+        if (count($lpItems) != 0) {
+            $registeredUsers = self::getCompanyLearnpathSubscription(
+                $startDate,
+                $endDate,
+                $whereInLp);
+            foreach ($registeredUsers as $student) {
+                $totalStudents = count($student);
+                for ($i = 0; $i < $totalStudents; $i++) {
```

Comment:
Usar foreach en lugar de for y cambiar el nombre de $student por $students o $studentList por que es un array.

Reply:
comentario añadido "use 'for' to performance"